controllers/login.js

```diff
@@ -87,19 +87,51 @@ router.all('/', (req, res, next) => {
 	});
 });
 
+const getIservOauthSystem = (schools) => {
+	for (let schoolIndex = 0; schoolIndex < schools.length; schoolIndex += 1) {
+		const { systems } = schools[schoolIndex];
+		for (let systemIndex = 0; systemIndex < systems.length; systemIndex += 1) {
+			if (systems[systemIndex].type === 'iserv') return systems[systemIndex];
+		}
+	}
+	return null;
+};
+
+const mapErrorcodeToTranslation = (errorCode) => {
+	if (errorCode === 'OauthLoginFailed') return 'login.text.oauth.loginFailed';
```

Comment:
are there any other cases?

Reply:
Not yet, but there will be.

Reply:
YAGNI & KISS

Reply:
ok... we marked the line as a comment, so we'll have the structure when we are going to need it.